docs/02-app/02-api-reference/04-functions/server-actions.mdx

```diff
@@ -0,0 +1,141 @@
+  }
+}
+```
+
+### With Client Components
+
+#### Import
+
+Create your Server Action in a separate file with the `"use server"` directive at the top of the file. Then, import the Server Action into your Client Component:
+
+```js filename="app/actions.js" highlight={1}
+'use server'
+
+export async function myAction() {
+  // ...
+}
+```
+
+```jsx filename="app/client-component.jsx" highlight={1}
+'use client'
+
+import { myAction } from './actions'
+
+export default function ClientComponent() {
+  return (
+    <form action={myAction}>
+      <button type="submit">Add to Cart</button>
+    </form>
+  )
+}
+```
+
+> **Good to know**: When using a top-level `"use server"` directive, all exports below will be considered Server Actions. You can have multiple Server Actions in a single file.
+
+#### Props
+
+In some cases, you might want to pass down a Server Action to a Client Component as a prop.
+
+```jsx
+<ClientComponent updateItem={updateItem} />
+```
+
+```jsx filename="app/client-component.jsx"
+'use client'
+
+export default function ClientComponent({ myAction }) {
+  return (
+    <form action={myAction}>
+      <input type="text" name="name" />
+      <button type="submit">Update Item</button>
+    </form>
+  )
+}
+```
+
+## Invocation
+
+You can invoke Server Actions using the following methods:
+
+- Using `action`: React's `action` prop allows invoking a Server Action on a `<form>` element.
+- Using `formAction`: React's `formAction` prop allows handling `<button>`, `<input type="submit">`, and `<input type="image">` elements in a `<form>`.
+- Custom Invocation with `startTransition`: Invoke Server Actions without using `action` or `formAction` by using `startTransition`. This method **disables [Progressive Enhancement](#progressive-enhancement)**.
+
+## Progressive Enhancement
+
+Progressive enhancement allows a `<form>` to function properly without JavaScript, or with JavaScript disabled. This allows users to interact with the form and submit data even if the JavaScript for the form hasn't been loaded yet or if it fails to load.
+
+React Actions (both server and client) support progressive enhancement, using one of two strategies:
+
+- If a **Server Action** is passed directly to a `<form>`, the form is interactive **even if JavaScript is disabled**.
+- If a **Client Action** is passed to a `<form>`, the form is still interactive, but the action will be placed in a queue until the form has hydrated. React prioritizes the action, so it still happens quickly.
+
+In both cases, the form is interactive before hydration occurs. Although Server Actions have the additional benefit of not relying on client JavaScript, you can still compose additional behavior with Client Actions where desired without sacrificing interactivity.
+
+## Size Limitation
+
+By default, the maximum size of the request body sent to a Server Action is 1MB. This prevents large amounts of data being sent to the server, which consumes a lot of server resource to parse.
```

Comment:
I got a little tripped up when reading this, some suggestions (but feel free to ignore):

```suggestion
By default, the maximum size of the request body sent to a Server Action is 1MB. This prevents large amounts of data from being sent to the server, which consumes a lot of server resources to parse.
```

Or maybe just:

```suggestion
By default, the maximum size of the request body sent to a Server Action is 1MB to prevent the consumption of excessive server resources in parsing large amounts of data.
```

Reply:
Thank you!